src/Raven.Server/Documents/Sharding/Handlers/Processors/TimeSeries/ShardedTimeSeriesHandlerProcessorForGetTimeSeries.cs

```diff
@@ -18,24 +24,65 @@ public ShardedTimeSeriesHandlerProcessorForGetTimeSeries([NotNull] ShardedDataba
         {
             var shardNumber = RequestHandler.DatabaseContext.GetShardNumber(context, docId);
+            
+            Action<ITimeSeriesIncludeBuilder> action = includeBuilder =>
+            {
+                if (includeDoc)
+                    includeBuilder.IncludeDocument();
+                if (includeTags)
+                    includeBuilder.IncludeTags();
+            };
+            
+            var cmd = new GetTimeSeriesOperation.GetTimeSeriesCommand(docId, name, from, to, start, pageSize, action, fullResults);
+            var rangeResult = await RequestHandler.ShardExecutor.ExecuteSingleShardAsync(context, cmd, shardNumber);
+
+            if (rangeResult == null)
+            {
+                if(cmd.StatusCode == HttpStatusCode.NotFound)
+                    RequestHandler.HttpContext.Response.StatusCode = (int)cmd.StatusCode;
+                return null;
+            }
+
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "Handle not modified for include tags");
 
-            if (includeTags)
+            var nonLocalIncludes = new List<string>();
+            foreach (var id in rangeResult.MissingIncludes)
             {
-                DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "Support include tags in time series?");
-                throw new NotSupportedInShardingException("Include tags of time series is not supported in sharding");
+                if(RequestHandler.DatabaseContext.GetShardNumber(context, id) != shardNumber)
+                    nonLocalIncludes.Add(id);
             }
 
-            Action<ITimeSeriesIncludeBuilder> builder = null;
-            if (includeDoc)
-                builder = bldr => bldr.IncludeDocument();
+            var idsByShards = ShardLocator.GetDocumentIdsByShards(context, RequestHandler.DatabaseContext, nonLocalIncludes);
+            var fetchDocsOp = new FetchDocumentsFromShardsOperation(context, RequestHandler, idsByShards, etag: null, includePaths: null, metadataOnly: false);
 
-            var cmd = new GetTimeSeriesOperation.GetTimeSeriesCommand(docId, name, from, to, start, pageSize, builder, fullResults);
-            var proxy = new ProxyCommand<TimeSeriesRangeResult<TimeSeriesEntry>>(cmd, RequestHandler.HttpContext.Response);
-            await RequestHandler.ShardExecutor.ExecuteSingleShardAsync(proxy, shardNumber);
+            ShardedReadResult<GetShardedDocumentsResult> result;
+            using (var token = RequestHandler.CreateOperationToken())
+                result = await RequestHandler.ShardExecutor.ExecuteParallelForShardsAsync(idsByShards.Keys.ToArray(), fetchDocsOp, token.Token);
+            
+            var includesDocId = $"TimeSeriesRangeIncludes/{docId}";
+            using (context.OpenReadTransaction())
+            {
+                rangeResult.Includes ??= context.ReadObject(new DynamicJsonValue(), includesDocId);
```

Comment:
wouldn't it be better to create one djv at the beginning, add all the data to it and readObject it at the end?